app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
> or are you concerned about having a single controller handling both standard and API requests?

@oeoeaio yep, I would use 2 different controllers and move the shared logic (if any) in some service. Would be easier to test, understand and debug IMHO.